crates/core/src/delta_datafusion/cdf/scan.rs

```diff
@@ -0,0 +1,59 @@
+use std::any::Any;
+use std::fmt::Formatter;
+use std::sync::Arc;
+
+use arrow_schema::SchemaRef;
+use datafusion::execution::{SendableRecordBatchStream, TaskContext};
+use datafusion::physical_plan::{DisplayAs, DisplayFormatType, ExecutionPlan};
+
+/// Physical execution of a scan
+#[derive(Debug, Clone)]
+pub struct DeltaCdfScan {
+    plan: Arc<dyn ExecutionPlan>,
+}
+
+impl DeltaCdfScan {
+    /// Creates a new scan
+    pub fn new(plan: Arc<dyn ExecutionPlan>) -> Self {
+        Self { plan }
+    }
+}
+
+impl DisplayAs for DeltaCdfScan {
+    fn fmt_as(&self, _t: DisplayFormatType, f: &mut Formatter) -> std::fmt::Result {
+        write!(f, "{:?}", self)
+    }
+}
+
+impl ExecutionPlan for DeltaCdfScan {
+    fn as_any(&self) -> &dyn Any {
+        self
+    }
+
+    fn schema(&self) -> SchemaRef {
+        self.plan.schema().clone()
+    }
+
+    fn properties(&self) -> &datafusion::physical_plan::PlanProperties {
+        self.plan.properties()
+    }
+
+    fn children(&self) -> Vec<Arc<dyn ExecutionPlan>> {
+        vec![]
+    }
+
+    fn with_new_children(
+        self: Arc<Self>,
+        _children: Vec<Arc<dyn ExecutionPlan>>,
```

Comment:
nit: you do use this, so why `_children` and not `children`?

Reply:
A side effect of me having this stubbed to start as `todo!()` and then not fixing it properly.